mantidproject/mantid
# Muon algorithms and dead detectors

Multiple muon algorithms contained the same bug, which caused them to produce nonsense data when the input workspace contained a dead detector (i.e. zero counts for all time). This PR fixes the algorithms. 


**To test:**

To test RemoveExpDecay: 

1. Load MUSR00069006 from the load button
2. Open the data table and note that spectrum 24 is empty (all zeros)
3. Run RemoveExpDecay
4. Check the output is all zeros for spectrum 24 (previously this was not the case).

To test PhaseQuad and CalMuonDetectorPhases

1. Open Muon Analysis
2. Load MUSR00069006 in Muon Analysis
3. Open the frequency domain analysis
4. Change the workspace to PhaseQuad
5. Press calculate FFT
6. Open the PhaseTable and check that detector 24 has an asymmetry of 999 and a phase of 0
7. Expand the MUSR00069006 group and check the PhaseQuad workspace (this was previously all NAN's)

To test MuonMaxEnt (if not continuing from the above do steps 1-3 inclusive from the above)
1. Change the top drop down in the Frequency domain analysis to MaxEnt
2. Change the workspace to MUSR00069006
3. Tick all of the output options in the top table
4. Press calculate MaxEnt
5. More workspace will be added to the workspace group MUSR00069006
6. Check that the FrequencyDomain;Maxent is not NAN
7. Check that the PhaseTable;MaxEnt has asymmetry = 999 and phase =0 for detector 24
8. Check that the DeadTimeTable has a value of zero for detector 24
9. Check that the TimeDomain and PhaseConvergenceTable both have zeros for spectrum 24



Fixes #22294. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
